### PR TITLE
Refactor NodeBalancer Redux state

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Includes both NodeBalancer Configs and NodeBalancerConfig Nodes. The Nodes have been in Redux for a while now, but as far as I know have never been used.

NodeBalancer actions should be unchanged.

### CODE REVIEW

File: packages/manager/src/store/store.helpers.tmp.ts [hunk 1]
```diff
@@ -98,7 +98,8 @@ export const addMany = <E extends Entity, O = APIError[] | undefined>(
   return {
     ...state,
     itemsById,
-    results: results ?? Object.keys(itemsById).length
+    results: results ?? Object.keys(itemsById).length,
+    lastUpdated: Date.now()
```
Comment (Jskobos) [inline packages/manager/src/store/store.helpers.tmp.ts:102]: Maybe there was a reason we weren't setting this before? Seems debatable.

File: packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.reducer.ts [hunk 2]
```diff
@@ -24,99 +23,162 @@ import {
...
-      return {
-        ...state,
-        loading: false,
-        lastUpdated: Date.now()
-      };
+  return produce(state, draft => {
+    if (isType(action, getAllNodeBalancerConfigsActions.started)) {
+      const { nodeBalancerId } = action.payload;
+      draft = ensureInitializedNestedState(draft, nodeBalancerId, {
+        results: 0
```
Comment (johnwcallahan) [inline packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.reducer.ts:38]: What's the `results: 0` pattern about?

File: packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.requests.ts [hunk 3]
```diff
@@ -83,11 +82,16 @@ export const updateNodeBalancerConfigs: ThunkActionCreator<
     const [added, removed] = getAddRemoved(storedConfigs, data);
 
     if (removed.length > 0) {
-      dispatch(removeNodeBalancerConfigs(removed.map(({ id }) => id)));
+      dispatch(
```
Comment (johnwcallahan) [inline packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.requests.ts:85]: A few lines above (L72) the old state shape is assumed, so I think the logic in this function won't work correctly.

File: packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.reducer.ts (hunk 2 above)
Comment (Jskobos) [reply packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.reducer.ts:38]: lol I don't know, it came over from an earlier use of ensureNestedDefaultState and I wasn't sure why it was there but didn't want to break anything.

File: packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.requests.ts (hunk 3 above)
Comment (Jskobos) [reply packages/manager/src/store/nodeBalancerConfig/nodeBalancerConfig.requests.ts:85]: This whole thing is unnecessary now I think, deleting it

**Pull request merged**